### DISCUSSION TITLE
Problem registering handler

### DISCUSSION THREAD

**Discussion by GonzaloMedinaMedina** [LuckyPennySoftware/MediatR#704] [Q&A]
Hi! I'm sending a message with mediator from ProjectA and my handler is implemented in ProjectB, but when trying to send the message there are no handlers registered.
Is there a way to register handlers without knowing where are located?

Regards!

**Comment by jbogard** [comment]
Aren't they in the same application? The application should register the handlers for A and B.

**Reply by GonzaloMedinaMedina** [reply]
Same application but different solutions. And solution A can not reference solution B.

**Reply by jbogard** [reply]
Is it the same application or separate applications? Those two sentences contradict each other.

**Reply by GonzaloMedinaMedina** [reply]
Sorry, I wrote wrong the sentence, one application built with few solutions.

**Reply by jbogard** [reply]
How does the one application load the other solutions? Do you have like a diagram that shows what the relationships are?

**Reply by GonzaloMedinaMedina** [reply]
The relation is made by dependency, the solution B has a dependency with A. I want to send a message from A and handle in B, but I can't reference code of project B in A. 
So there is a way to register a handle in A that is implemented in B? Something like passing a string path of the file to the scanner?

**Reply by jbogard** [reply]
The application needs to do all the registrations. If these applications are in the same process it should be no problem at all, you'd just do something like:

```csharp
services.AddMediatR(
   typeof(SomeTypeInA).Assembly,
   typeof(SomeTypeInB).Assembly,
   typeof(SomeTypeInC).Assembly,
   // etc
);
```

I don't understand why solutions can't reference each other, there's no technical reason why that can't happen. But that doesn't really matter here. If all of these projects/solutions are in the same application, and that application contains the single composition root for dependency injection, then you just have to register those handlers with the composition root. Not each project/solution with each other.

**Reply by GonzaloMedinaMedina** [reply]
There is a way to do these registers in a Mvc.net application?

**Reply by jbogard** [reply]
MVC 4/5? Yes, there's some docs in the wiki for registering directly against containers. You can also look at the samples folder, it has how to do different containers.

**Reply by GonzaloMedinaMedina** [reply]
Thank you but I found another problem, I'm using OwinStartup, there is a way to do this register through a IAppBuilder?
Because I can't register the service directly in the container like your screenshot.

**Reply by jbogard** [reply]
No idea, I've not used that in quite a long time. Here's what I found on StackOverflow:

https://stackoverflow.com/questions/19781970/how-to-use-di-container-when-owinstartup
